File: backend/StudHunter.DB.Postgres/Models/Student.cs
namespace StudHunter.DB.Postgres.Models;

public class Student : User
{
    private string _firstName = null!;
    public string FirstName
    {
        get => _firstName;
        set
        {
            ArgumentException.ThrowIfNullOrWhiteSpace(value);
            _firstName = value.Trim();
        }
    }
    private string _lastName = null!;
    public string LastName
    {
        get => _lastName;
        set
        {
            ArgumentException.ThrowIfNullOrWhiteSpace(value);
            _lastName = value.Trim();
        }
    }

    private string? _patronymic;
    public string? Patronymic
    {
        get => _patronymic;
        set => _patronymic = value?.Trim();
    }

    public StudentGender? Gender { get; set; }
    public DateOnly? BirthDate { get; set; }
    public bool? IsForeign { get; set; }
    public StudentStatus Status { get; set; } = StudentStatus.Studying;

    public virtual StudyPlan? StudyPlan { get; set; }
    public virtual Resume? Resume { get; set; }

    public enum StudentGender
    {
        Male,
        Female
    }

    public enum StudentStatus
    {
        Studying = 1,
        SeekingInternship = 2,
        SeekingJob = 3,
        Interning = 4,
        Working = 5
    }
}

File: frontend/src/types/student.ts
export interface DictionaryItem {
  id: string
  name: string
}

export interface StudentProfileDto {
  id: string
  email: string
  firstName: string
  lastName: string
  patronymic: string | null
  contactPhone: string | null
  contactEmail: string | null
  cityId: string | null
  gender: 'Male' | 'Female' | null
  birthDate: string | null
  avatarUrl: string | null
  isForeign?: boolean | null
  status: 'Studying' | 'SeekingJob' | 'SeekingInternship' | 'Interning' | 'Working'
  universityId: string | null
  facultyId: string | null
  departmentId: string | null
  studyDirectionId: string | null
  courseNumber: number
  studyForm: 'FullTime' | 'PartTime' | 'Correspondence'
  courseIds: string[]
  courses: { id: string; name: string }[]
}

export interface ResumeFillDto {
  id?: string | null
  title: string
  description: string | null
  isDeleted: boolean
  skillIds: string[]
  skills: { id: string; name: string }[]
  updatedAt?: string | null
}

export interface VacancyShortDto {
  id: string
  title: string
}

export interface StudentHeroDto {
  fullName: string
  birthDate: string | null
  avatarUrl: string | null
  status: string
  universityName: string | null
  facultyName: string | null
  departmentName: string | null
  studyDirectionName: string | null
  courseNumber: number | null
}
